zb-db/src/main/java/io/zeebe/db/impl/rocksdb/ZeebeRocksDbFactory.java

```diff
@@ -103,6 +104,13 @@ private DBOptions createDefaultDbOptions(final List<AutoCloseable> closeables) {
             .setLogFileTimeToRoll(Duration.ofMinutes(30).toSeconds())
             .setKeepLogFileNum(2);
 
+    // limit I/O writes
+    if (rocksDbConfiguration.getIoRateBytesPerSecond() > 0) {
+      final RateLimiter rateLimiter =
+          new RateLimiter(rocksDbConfiguration.getIoRateBytesPerSecond());
+      dbOptions.setRateLimiter(rateLimiter);
+    }
```

Comment:
This replaces bytesPerSync?

Reply:
Not quite - regarding bytesPerSync, I found it had simply no effect at all. You can look at the `np-no-fsync` benchmark. That was setting when you would call fsync (or flush), so it just ensured you fsync'd every 4MB, regardless of how fast you were writing those. 

This new setting is intended to limit how much you can write per second, so it could smooth out I/O performance, which could help having more predictable performance.